org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/properties/SonarLintProjectPropertyPage.java

```diff
@@ -135,14 +135,14 @@ private void updateState() {
     Optional<EclipseProjectBinding> projectBinding = getProjectConfig().getProjectBinding();
     if (projectBinding.isPresent()) {
       boundDetails
-        .setText("Bound to the project '" + projectBinding.get().projectKey() + "' on server '" + serverName(projectBinding.get().serverId()) + "'");
+        .setText("Bound to the project '" + projectBinding.get().projectKey() + "' on server '" + serverName(projectBinding.get().connectionId()) + "'");
       bindLink.setText("<a>Update project binding</a>");
     } else {
       bindLink.setText("<a>Bind this Eclipse project to a SonarQube project</a>");
       boundDetails.setText("");
     }
     if (projectBinding.isPresent() && !SonarLintCorePlugin.getServersManager().forProject(getProject()).isPresent()) {
-      addServerLink.setText("<a>Connect to SonarQube server '" + projectBinding.get().serverId() + "'</a>");
+      addServerLink.setText("<a>Connect to SonarQube server '" + projectBinding.get().connectionId() + "'</a>");
```

Comment:
Found another lone "SonarQube" here

Reply:
This text is shown when the project is bound but when the binding is 'broken', that is when the connection has been deleted but the project is still bound. If we click on the link, we open the popup to create a new connection, prefilled with the connectionId. 

I changed the text to 'Re-create SonarQube/SonarCloud connection ...'. Because Connect was not the right term here. 

Still I am wondering: this happens probably because the user deleted a connection but didn't unbind the project. When we delete a connection, couldn't we also unbind all the projects bound to this connection ? After informing user of course

Reply:
This is also the reason why I introduced `ResolvedBinding` in another commit. When a project has a binding, in a lot of places we have to check if the connection is still there, thus the "resolve" operation, which is maybe a poor term.

To summarize, today we have 3 states for a project: unbound, bound to an existing connection, bound to a missing connection. I think we could simplify to keep the 2 first states. If the 3rd happens because of an error, we could remove the binding

Reply:
This situation can happen when users share project settings (either in SCM or using workspace mechanics). The project settings contain the binding with a connection id, but the user misses the connection definition. I don't think we should ignore this case.